lib/gui/app/components/file-selector/file-selector/file-list.jsx

```diff
@@ -0,0 +1,318 @@
+  }
+
+  onSelect (event) {
+    event.preventDefault()
+    this.props.onSelect(this.props.file)
+  }
+
+  render () {
+    const file = this.props.file
+    return (
+      <ClickableFlex
+        data-path={ file.path }
+        href={ `file://${file.path}` }
+        direction="column"
+        alignItems="stretch"
+        className={ this.props.className }
+        onClick={ ::this.onHighlight }
+        onDoubleClick={ ::this.onSelect }>
+        <span className={ UnstyledFile.getFileIconClass(file) } />
+        <span>{ middleEllipsis(file.basename, FILENAME_CHAR_LIMIT_SHORT) }</span>
+        <div>{ file.isDirectory ? '' : prettyBytes(file.stats.size || 0) }</div>
+      </ClickableFlex>
+    )
+  }
+}
+
+/**
+ * @summary File element
+ * @class
+ * @type {StyledComponent}
+ */
+const File = styled(UnstyledFile)`
+  width: 100px;
+  min-height: 100px;
+  max-height: 128px;
+  margin: 5px 10px;
+  padding: 5px;
+  background-color: none;
+  transition: 0.05s background-color ease-out;
+  color: ${ colors.primary.color };
+  cursor: pointer;
+  border-radius: 5px;
+  word-break: break-word;
+
+  :hover, :visited {
+    color: ${ colors.primary.color };
+  }
+
+  :focus,
+  :active {
+    color: ${ colors.highlight.color };
+    background-color: ${ colors.highlight.background };
+  }
+
+  :focus > span:first-of-type,
+  :active > span:first-of-type {
+    color: ${ colors.highlight.color };
+  }
+
+  :focus > div:last-child,
+  :active > div:last-child {
+    color: ${ colors.highlight.color };
+  }
+
+  > span:first-of-type {
+    align-self: center;
+    line-height: 1;
+    margin-bottom: 6px;
+    font-size: 48px;
+    color: ${ props => props.disabled ? colors.primary.subColor : colors.soft.color };
+  }
+
+  > span:last-of-type {
+    display: flex;
+    justify-content: center;
+    text-align: center;
+    font-size: 14px;
+  }
+
+  > div:last-child {
+    background-color: none;
+    color: ${ colors.primary.subColor };
+    text-align: center;
+    font-size: 12px;
+  }
+`
+
+/**
+ * @summary FileList element
+ * @class
+ * @type {ReactElement}
+ */
+class FileList extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      path: props.path,
+      highlighted: null,
+      files: [],
+    }
+  }
+
+  readdir (dirname) {
+    debug('FileList:readdir', dirname)
+    files.readdirAsync(dirname).then((files) => {
+      window.requestAnimationFrame(() => {
+        this.setState({ files: files })
+      })
+    })
+  }
+
+  componentDidMount () {
+    process.nextTick(() => {
+      this.readdir(this.state.path)
+    })
+  }
+
+  onHighlight (file) {
+    debug('FileList:onHighlight', file)
+    // this.setState({ highlighted: file.path })
+    this.props.onHighlight(file)
+  }
+
+  onSelect (file) {
+    debug('FileList:onSelect', file.path, file.isDirectory)
+    this.props.onSelect(file)
+  }
+
+  componentDidUpdate () {
+    console.timeEnd('FileList:componentDidUpdate')
```

Comment:
Shouldn't we use `debug` for that?

Reply:
`debug` doesn't support timing like this that mark timings in the devtools' perf timeline